src/units/attack_type.hpp

```diff
@@ -134,7 +134,7 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 	 * @param filter_self name of [filter_"self/student"] if is abilities or specials who are checked
 	 * @param is_special for determine if list is a special or a ability.
 	 */
-	unit_ability_list overwrite_special_checking(const std::string& ability, const unit_ability_list& temp_list, const unit_ability_list& abil_list, const std::string& filter_self, bool is_special) const;
+	unit_ability_list overwrite_special_checking(const std::string& ability, unit_ability_list temp_list, unit_ability_list abil_list, const std::string& filter_self, bool is_special) const;
```

Comment:
I found the documentation a bit confusing, and think this would be clearer (including renaming some of the arguments). Please would you check if it's correct?

Assuming it is correct, you can either cherry-pick it or I can create a PR for it afterwards.

```
	/**
	 * Filter a list of abilities or weapon specials, removing any entries that are overridden by
	 * the overwrite_specials attributes of a second list.
	 *
	 * @param ability The special ability type who is being checked.
	 * @param input list to check, a filtered copy of this list is returned by the function.
	 * @param overwriters list that may have overwrite_specials attributes.
	 * @param filter_self name of [filter_"self/student"] if is abilities or specials who are checked.
	 * @param is_special if true, input contains weapon specials; if false, it contains abilities.
	 */
	unit_ability_list overwrite_special_checking(const std::string& ability, unit_ability_list input, unit_ability_list overwriters, const std::string& filter_self, bool is_special) const;
```